integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+          var dev_count;
+          var user_count;
+	  //check each device in our expected list
+          for (dev_count=0; dev_count < expectedDevices.length; dev_count++) {    
+            var found=false;
+	    // is found somewhere i the actual list
+	    devices.forEach(function(device) {
+	      // and contains the correct values
+              if(device.devicename==expectedDevices[dev_count].devicename) {
+		found=true;
+		expect(device.id).to.equal(expectedDevices[dev_count].id);
+                expect(device.devicename).to.equal(expectedDevices[dev_count].devicename);
+                expect(device.active).to.equal(expectedDevices[dev_count].active);
+                if(expectedDevices[dev_count].Users==null) {
+                  expect(device.Users).to.not.exist;
+                } else {
+                  expect(device.Users.length).to.equal(expectedDevices[dev_count].Users.length);
+	          // sort users and expected users to ensure order is the same
+		  var users=sortArrayOn(device.Users,'username');
+		  var expectedUsers=sortArrayOn(expectedDevices[dev_count].Users,'username');
+
+	          // compare user list
+                  var count;
+                  for (count=0; count < expectedDevices[dev_count].Users.length; count++) {
+                      expect(users[count].username).to.equal(expectedUsers[count].username);
+                      expect(users[count].id).to.equal(expectedUsers[count].id);
+                      expect(users[count].DeviceUsers.admin).to.equal(expectedUsers[count].DeviceUsers.admin);
+                      expect(users[count].DeviceUsers.UserId).to.equal(expectedUsers[count].DeviceUsers.UserId);
+                  };
+                };
+              };
+	    });
+            expect(found).to.equal(true);
+          }
+       };
+  });
+});
+
+Cypress.Commands.add("apiCheckDeviceInGroup", (userName: string, cameraName: string, groupName: string, groupId: number, expectedDevice: all, params:string  = {}, statusCode: number = 200) => {
+  logTestDescription(
+      `${userName} Check user '${userName}' can see device '${cameraName}' in group '${groupName}' `,
+      { user: userName, groupName, cameraName },
+      true
+  );
+
+  // use group id if present, otherwise query by name
+  var fullUrl = null;
+  if(groupId!=null) {
+    fullUrl = v1ApiPath('devices/'+getTestName(cameraName)+'/in-group/'+groupId);
+  } else {
+    fullUrl = v1ApiPath('devices/'+getTestName(cameraName)+'/in-group/'+getTestName(groupName));
+  };
+
+    logTestDescription(`Check that ${userName} get device ${cameraName} in group ${groupName} returns ${statusCode} and correct data`, {});
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "GET",
+        url: fullUrl,
+        body: null
+      },
+      userName,
+      statusCode
+    ).then((response) => {
+       if(statusCode==null || statusCode==200) {
+	  var device=response.body.device;
+	  expect(device.id).to.equal(getCreds(cameraName).id);
```

Comment:
i'd say these comparison of types, could be put into seperate functions and referenced in all places used

Reply:
Noted

Reply:
Noted.  But currently there is no consistency of these types between different API endpoints.  So there is no opportunity for reuse.  Standardise the types and we can standardise the testing.

Reply:
types all now declared in types.ts - so code can be reused if we identify any types that are reused